pkg/ingester/wal.go

```diff
@@ -49,15 +51,36 @@ func (cfg *WALConfig) RegisterFlags(f *flag.FlagSet) {
 // WAL interface allows us to have a no-op WAL when the WAL is disabled.
 type WAL interface {
 	// Log marshalls the records and writes it into the WAL.
-	Log(*Record) error
+	Log(*WALRecord) error
 	// Stop stops all the WAL operations.
 	Stop()
 }
 
+// RecordType represents the type of the WAL/Checkpoint record.
+type RecordType byte
+
+const (
+	// Currently we also support the old records without a type header.
+	// For that, we assume the record type does not cross 7 as the proto unmarshalling
+	// will produce an error if the first byte is less than 7 (thus we know its not the old record).
+	// The old record will be removed in the future releases, hence the record type should not cross
+	// '7' till then.
+	// Relevant code to verify this is https://github.com/cortexproject/cortex/blob/90516410b9e1fe2fe3ae9041260c02d2a6f0468d/pkg/ingester/wal.pb.go#L894-L917
+	// (with error at line 916).
+
+	// WALRecordSeriesType1 is the type for the WAL record on Prometheus TSDB record for series.
+	WALRecordSeriesType1 RecordType = 1
+	// WALRecordSamplesType1 is the type for the WAL record based on Prometheus TSDB record for samples.
+	WALRecordSamplesType1 RecordType = 2
```

Comment:
What is `Type1` suffix?

Reply:
I just meant to say first one in the series of records for samples. Similarly, if we plan to change that record, it would be `Type2` (with the description in comments).  What other name do you suggest?

Reply:
I would just remove `Type1` until we need `Type2` :)

Reply:
(I mean the suffix in the name)